src/newlines.cpp

```diff
@@ -3231,7 +3231,7 @@ void newlines_chunk_pos(c_token_t chunk_type, tokenpos_e mode)
    chunk_t *prev;
    int     nl_flag;
 
-   if ((mode & (TP_JOIN | TP_LEAD | TP_TRAIL)) == 0)
+   if ((mode & (TP_JOIN | TP_LEAD | TP_TRAIL)) == 0 && chunk_type != CT_COMMA)
```

Comment:
Not entirely. I was just thinking about showing how the previous code was wrong for the UO_pos_class_comma but if you say that it's a different code path (handled in `newlines_class_colon_pos` I guess that's not possible to show.